xl/player/gst/engine.py

```diff
@@ -542,11 +544,14 @@ def seek(self, value):
 
         return self.playbin.send_event(seek_event)
 
-    def set_volume(self, volume):
+    def set_volume(self, volume: float) -> None:
         # self.logger.debug("Set playbin volume: %.2f", volume)
         # TODO: strange issue where pulse sets the system audio volume
         #       when exaile starts up...
-        self.playbin.props.volume = volume
+        if not self.needs_sink:
+            self.audio_sink.set_volume(volume)
+        else:
+            self.playbin.props.volume = volume
```

Comment:
This is a bit unsettling to me. I feel like it should either *all* be managed by the playbin volume, or never managed by the playbin volume.

I'm not sure if this could ever happen, but if someone set the volume to 50% when the audio sink wasn't present, once you add the audio sink the playbin volume can never be adjusted.

Reply:
This happens only right after starting exaile before playing the first track. To me it seems that if I set the playbin volume before configuring the sink, the sink volume is set to this volume after configuring. But may be I am wrong because of https://github.com/exaile/exaile/issues/709.
We can keep the volume in the engine and set volume of the sink right when it is present.

I saw there is `on_volume_change` that is also reading the volume from the playbin.

Reply:
I think if you never touch volume in playbin that would be best.